src/MCPClient/tests/test_verify_checksum.py

```diff
@@ -0,0 +1,285 @@
+# -*- coding: utf-8 -*-
+
+"""Test Verify Checksum Job in Archivematica.
+
+Tests for the verify checksum Job in Archivematica which makes calls out to the
+hashsum checksum utilities. We need to ensure that the output of the tool is
+mapped consistently to something that can be understood by users when
+debugging their preservation workflow.
+"""
+
+# This file is part of Archivematica.
+#
+# Copyright 2010-2017 Artefactual Systems Inc. <http://artefactual.com>
+#
+# Archivematica is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Archivematica is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Archivematica.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+import os
+import subprocess
+import sys
+from uuid import UUID
+
+from django.core.management import call_command
+
+import pytest
+
+from main.models import File, Event
+
+THIS_DIR = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(os.path.abspath(os.path.join(THIS_DIR, "../lib/clientScripts")))
+
+from job import Job
+from verify_checksum import (
+    Hashsum,
+    NoHashCommandAvailable,
+    write_premis_event_per_file,
+    PREMISFailure,
+)
+
+
+class TestHashsum(object):
+    """Hashsum test runner object."""
+
+    @staticmethod
+    def setup_hashsum(path, job):
+        """Return a hashsum instance to calling functions and perform any
+        other additional setup as necessary.
+        """
+        return Hashsum(path, job)
+
+    def test_invalid_initialisation(self):
+        """Test that we don't return a Hashsum object if there isn't a tool
+        configured to work with the file path provided.
+        """
+        try:
+            Hashsum("checksum.invalid_hash")
+            assert False
+        except NoHashCommandAvailable:
+            assert True
```

Comment:
Not sure if our pytest version supports it but this could be more clear using [pytest.raises](https://docs.pytest.org/en/3.0.3/assert.html#assertions-about-expected-exceptions)